pkg/kf/commands/utils/prefix_writer.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+)
+
+type prefixWriter struct {
+	nw newlineWriter
+}
+
+func NewPrefixWriter(prefix string, dest io.Writer) io.Writer {
+	return &prefixWriter{
+		nw: newlineWriter{
+			dest: func(data []byte) (int, error) {
+				n, err := fmt.Fprintf(dest, "%s%s\n", prefix, data)
+				if err != nil {
+					return 0, err
+				}
+				return n - len(prefix), nil
+			},
+		},
+	}
+}
+
+func (w *prefixWriter) Write(data []byte) (int, error) {
+	return w.nw.Write(data)
+}
+
+type newlineWriter struct {
+	dest func([]byte) (int, error)
+	buf  bytes.Buffer
+}
+
+func (w *newlineWriter) Write(data []byte) (int, error) {
+	r := bufio.NewReader(bytes.NewReader(data))
```

Comment:
```suggestion
func (w *newlineWriter) Write(data []byte) (int, error) {
	w.buf.Write(data)

	for {
		line, err := w.buf.ReadBytes('\n')
		// From the docs: ReadBytes returns err != nil if and only if the returned
		// data does not end in delim and Buffer will not error unless it's empty.
		if err != nil {
			break
		}

		w.dest(line)
	}

	return len(data), nil
}
```

Reply:
This does not handle the case where multiple writes are used for a single line:

```
f.Write([]byte("[prefix-1] "))
f.Write([]byte("data-0\n"))
```